[BUG] ‘::rsqrt’ has not been declared when building MatX

**Describe the Bug**
A clear and concise description of what the bug is.

I include `matx.h` at the top of my file, and I get an error saying that `rsqrt` has not been declared.

This was following a clean build and install of the latest MatX changes (from commit #703), but don't know which commit caused it. But before pulling I was up to commit #8b8da5a0

**To Reproduce**
Steps to reproduce the behavior:
Basically I pulled the latest changes commit #e2cb0de (PR #703) built my tests and got the trace:

```
In file included from /usr/local/include/matx/core/type_utils.h:44,
                 from /usr/local/include/matx/core/pybind.h:34,
                 from /usr/local/include/matx/core/file_io.h:47,
                 from /usr/local/include/matx.h:41,
                 from /home/hphi344/Documents/GS-DBSCAN-CPP/test/../include/TestUtils.h:14,
                 from /home/hphi344/Documents/GS-DBSCAN-CPP/test/TestUtils.cpp:6:
/usr/local/include/matx/core/half.h: In function ‘matx::matxHalf<T> matx::rsqrt(const matx::matxHalf<T>&)’:
/usr/local/include/matx/core/half.h:704:27: error: ‘::rsqrt’ has not been declared
  704 |   return static_cast<T>(::rsqrt(static_cast<float>(x.x)));
 ```

**Expected Behavior**
Build with no errors

**Code Snippets**
If applicable, add code snippets to help explain your problem.

**System Details (please complete the following information):**
 - OS: Ubuntu 20.04
 - CUDA version: CUDA 12.5
 - g++ version: 9.4.0

**Additional Context**
May be a config issue on my behalf, but I'd be keen to see if you guys can reproduce it.

--- a/include/matx/core/half.h
+++ b/include/matx/core/half.h
@@ -701,7 +701,11 @@ __MATX_HOST__ __MATX_DEVICE__ __MATX_INLINE__ matxHalf<T> rsqrt(const matxHalf<T
 #ifdef __CUDA_ARCH__
   return hrsqrt(x.x);
 #else
-  return static_cast<T>(::rsqrt(static_cast<float>(x.x)));
+  #ifdef __CUDACC__
+    return static_cast<__nv_bfloat16>(::rsqrt(static_cast<float>(x.x)));
+  #else
+    return static_cast<__nv_bfloat16>(1.f / cuda::std::sqrt(static_cast<float>(x.x)));
+  #endif
 #endif
 }
 
@@ -719,7 +723,11 @@ rsqrt(const matxHalf<__nv_bfloat16> &x)
 #if __CUDA_ARCH__ >= 800
   return hrsqrt(x.x);
 #else
-  return static_cast<__nv_bfloat16>(::rsqrt(static_cast<float>(x.x)));
+  #ifdef __CUDACC__
+    return static_cast<__nv_bfloat16>(::rsqrt(static_cast<float>(x.x)));
+  #else
+    return static_cast<__nv_bfloat16>(1.f / cuda::std::sqrt(static_cast<float>(x.x)));
+  #endif
 #endif
 }
 

--- a/include/matx/core/storage.h
+++ b/include/matx/core/storage.h
@@ -282,7 +282,7 @@ namespace matx
      * @brief Default construct a smart_pointer_buffer. This should only be used when temporarily
      * creating an empty tensor for construction later.
     */
-    smart_pointer_buffer<T>() {};
+    smart_pointer_buffer() {};
 
     /**
      * @brief Construct a new smart pointer buffer from an existing object
@@ -290,7 +290,7 @@ namespace matx
      * @param ptr Smart poiner object
      * @param size Size of allocation
      */
-    smart_pointer_buffer<T>(T &&ptr, size_t size) : data_(std::forward<T>(ptr)), size_(size) {
+    smart_pointer_buffer(T &&ptr, size_t size) : data_(std::forward<T>(ptr)), size_(size) {
       static_assert(is_smart_ptr_v<T>);
     }
 
@@ -328,7 +328,7 @@ namespace matx
      * 
      * @param rhs Object to move from
      */
-    smart_pointer_buffer<T>(smart_pointer_buffer<T> &&rhs) {
+    smart_pointer_buffer(smart_pointer_buffer<T> &&rhs) {
       size_ = rhs.size_; 
       data_ = std::move(rhs.data_);
     }    
@@ -337,7 +337,7 @@ namespace matx
      * @brief Default destructor
      * 
      */  
-    ~smart_pointer_buffer<T>() = default;
+    ~smart_pointer_buffer() = default;
 
     /**
      * @brief Get underlying data pointer

--- a/include/matx/operators/scalar_ops.h
+++ b/include/matx/operators/scalar_ops.h
@@ -162,7 +162,11 @@ static __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ auto _internal_rsqrt(T v1)
     return rsqrt(v1);
   }
   else {
+#ifdef __CUDACC__
     return ::rsqrt(v1);
+#else
+    return static_cast<T>(1) / sqrt(v1);
+#endif
   }
 }
 template <typename T> struct RSqrtF {

